CMakeLists.txt

```diff
@@ -40,6 +40,21 @@ else()
     set(ATOMIC_LIBS "")
 endif()
 
+# test whether we are building on a musl based system.
+# On these we need to increase the stack size as it is very low by default
+find_program(LDD_EXECUTABLE ldd)
+if(LDD_EXECUTABLE)
+    # we use /bin/ls as it is very likely to exist on any system. ldd prints the dynamic linked libraries
+    # of the given executable, if musl is used it will print it as a linked object
+    execute_process(COMMAND ${LDD_EXECUTABLE} /bin/ls OUTPUT_VARIABLE LDD_LS_OUTPUT ERROR_QUIET)
```

Comment:
Probably using find_program for `ls` would be needed as well, otherwise I could imagine this using a host binary in eg a cross compiling setup